### DISCUSSION TITLE
table customization

### DISCUSSION THREAD

**Discussion by marco-santulli** [Budibase/budibase#4881] [Ideas]
in the new feature to customize the table columns, I have the following behaviour:

i have a date, expressed as a long timestamp from the database being displayed in the following long format: March 1 2022, 00:00.  I would like to show it only as 01/03, or in any case the "european" way with DD/MM/YYY, but I can see I can modify the actual date, but not its formatting, because if I return something that is not a valid date I get an "Invalid Date" in the table.  Maybe it would be good if we were able to cast it to a string, I think that's what is currently happening to numbers because in that case I can format it without problems:

<img width="114" alt="image" src="https://user-images.githubusercontent.com/82715690/157711790-c6219807-5b41-4232-81a5-4f852f0ec556.png">

Marco

**Comment by mjashanks** [comment]
Hey @marco-santulli 

You can do this by adding a formula column to your data table - e.g. "Display Date". Then display this column instead of the actual date column.

**Reply by marco-santulli** [reply]
Hi Mike,

thx I will do it like that, but I think since the number column accepts formatted strings, maybe we should make the date column work the same way?

M

**Reply by mjashanks** [reply]
I see what you mean. Really, we should have a `Value` and a `Display` in this screen:

![image](https://user-images.githubusercontent.com/3524181/158152917-de6749b3-292c-45ef-9f08-91aad58207f6.png)

That would preserve the Value, so it could be sorted on

**Reply by marco-santulli** [reply]
good point, you are right.  
I guess the same would go for the other types like number type that currently accepts strings?
